InnoGotchiGame.Mvc/Controllers/FarmsController.cs

```diff
@@ -0,0 +1,186 @@
+﻿using Application.Services.DataTransferObjects.Reading;
+using InnoGotchiGame.Mvc.Models.Reading;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace InnoGotchiGame.Mvc.Controllers
+{
+    public class FarmsController : Controller
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public FarmsController(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        [HttpGet]
+        [Route("farms/overview")]
+        public async Task<IActionResult> FarmsOverview()
+        {
+            var jwtToken = Request.Cookies["jwtToken"];
+            var myFarmRequestMessage = new HttpRequestMessage(
+            HttpMethod.Get,
+            "https://localhost:44336/api/Farms/my-farm")
+            {
+                Headers =
+                {
+                    { HeaderNames.Accept, "application/json" },
+                    { HeaderNames.Authorization, $"Bearer {jwtToken}" }
+                }
+            };
+            var friendFarmsRequestMessage = new HttpRequestMessage(
+            HttpMethod.Get,
+            "https://localhost:44336/api/Farms/friends")
+            {
+                Headers =
+                {
+                    { HeaderNames.Accept, "application/json" },
+                    { HeaderNames.Authorization, $"Bearer {jwtToken}" }
+                }
+            };
+
+
+            var httpClient = _httpClientFactory.CreateClient();
+
+            var myFarmResponseMessage = await httpClient.SendAsync(myFarmRequestMessage);
+            if (myFarmResponseMessage.IsSuccessStatusCode)
+            {
+                var myFarmJson = await myFarmResponseMessage.Content.ReadAsStringAsync();
+                var myFarm = JsonConvert.DeserializeObject<FarmMinReadingDto>(myFarmJson);
+                ViewBag.MyFarm = myFarm;
+            }
+            else if (myFarmResponseMessage.StatusCode.Equals(HttpStatusCode.NotFound))
+                ViewBag.MyFarm = null;
+            else
+                return View("~/Views/Error.cshtml", myFarmResponseMessage.StatusCode.ToString());
+
+
+            var friendFarmsResponseMessage = await httpClient.SendAsync(friendFarmsRequestMessage);
+            if (!friendFarmsResponseMessage.IsSuccessStatusCode) 
+                return View("~/Views/Error.cshtml", friendFarmsResponseMessage.StatusCode.ToString());
+            var friendFarmsJson = await friendFarmsResponseMessage.Content.ReadAsStringAsync();
+            var friendFarms = JsonConvert.DeserializeObject<IEnumerable<FarmMinReadingDto>>(friendFarmsJson);
+            
+            ViewBag.FriendFarms = friendFarms;
```

Comment:
Why ViewBag instead of ViewData or Model?

Reply:
For me it is the most convenient one